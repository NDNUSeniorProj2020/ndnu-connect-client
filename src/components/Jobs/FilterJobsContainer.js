import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'antd';

import ListJobs from './ListJobs';

export default function FilterJobsContainer(props) {
	const [jobs, setJobs] = useState(props.jobs);
	const [jobTypes, setJobTypes] = useState([]);

	const filterJobs = typesStack => props.jobs.filter(job => typesStack.includes(job.type));

	const onChange = (e) => {		
		let typesStack = jobTypes;

		if (e.target.checked) {
			if (!(e.target.name in typesStack))
				typesStack.push(e.target.name);				
		} else {
			typesStack = typesStack.filter(type => type !== e.target.name);
		}

		// If typesStack is not empty, filter jobs. If it is empty, set jobs to the original jobs array
		setJobs(typesStack.length > 0 ? filterJobs(typesStack) : props.jobs);
		setJobTypes(typesStack);
	};

	return (
		<div>
			<Checkbox onChange={onChange} name="FULL" />Full-Time
			<Checkbox onChange={onChange} name="PART" />Part-Time
			<Checkbox onChange={onChange} name="INTERN" />Internship
			<ListJobs jobs={jobs} />
		</div>
	);
}

FilterJobsContainer.propTypes = { jobs: PropTypes.array };
FilterJobsContainer.defaultProps = { jobs: [] };
