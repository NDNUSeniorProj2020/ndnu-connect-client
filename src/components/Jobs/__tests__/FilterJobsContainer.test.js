import React from 'react';
import { shallow } from 'enzyme';

import FilterJobsContainer from '../FilterJobsContainer';

describe('tests for FilterJobsContainer', () => {
	describe('snapshot tests', () => {
		it('renders without crashing with no props passed', () => {
			const tree = shallow(<FilterJobsContainer />);
			expect(tree).toMatchSnapshot();
		});
	});
});
