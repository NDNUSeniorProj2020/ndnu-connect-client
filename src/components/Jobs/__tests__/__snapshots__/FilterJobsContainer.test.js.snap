// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for FilterJobsContainer snapshot tests renders without crashing with no props passed 1`] = `ShallowWrapper {}`;
